RecoJets/JetProducers/src/PileupJetIdAlgo.cc

```diff
@@ -367,7 +370,10 @@ PileupJetIdentifier PileupJetIdAlgo::computeIdVariables(const reco::Jet * jet, f
 			if( icone < ncones ) { *coneEmFracs[icone] += candPt; }
 			internalId_.ptDNe_    += candPt*candPt;
 			sumPtNe               += candPt;
+			multNeut += candPuppiWeight;
 		}
+		if((abs(icand->pdgId()) == 1) || (abs(icand->pdgId()) == 2)) multNeut += candPuppiWeight;
```

Comment:
What is the purpose of this line? PdgId 1,2 seem not to be used in PF? https://twiki.cern.ch/twiki/bin/view/CMSPublic/SWGuideParticleFlow#Particle_types

Reply:
These are HF had/gamma. The multNeut should reproduce patjet->neutralMultiplicity() exactly in the case that usePuppi is False.

So we have events like this (setting usePuppi to False)

Adding 130
Adding 130
Adding 22
Adding 22
Adding 22
Adding 22
Adding 22
Adding 22
multNeut = 8  ; patjet->neutralMultiplicity() = 8

and events like this 

Adding 1
Adding 1
Adding 1
Adding 1
Adding 1
multNeut = 5 ;  patjet->neutralMultiplicity() = 5

Reply:
ok makes sense!